pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -471,9 +472,54 @@ def pick_single_gpu(exclude_gpus: list):
     raise RuntimeError("No GPUs available.")
 
 
-def pick_multiple_gpus(nb):
+def pick_single_gpu_realist_workload(exclude_gpus: list, model: LightningModule) -> int:
+    batch = next(iter(model.train_dataloader))
+    for i in range(torch.cuda.device_count()):
+        if i in exclude_gpus:
+            continue
+        # Try to allocate on device:
+        device = torch.device(f"cuda:{i}")
+        try:
+            with torch.set_grad_enabled(True):
+                model_device = model.to(device)
+                batch_device = batch.to(device)
+                model_device.train()  # record grads
+                model_device(batch_device)
```

Comment:
this probably won't work, since usually a batch here is not only the input tensor, but can be of any type (including dicts and strings), which you don't want to feed into the network